cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -835,33 +921,15 @@ func (m *stateMachine) mountData() (stateFunc, error) {
 		return m.unlockSaveRunKey, nil
 	}
 
-	// if we are unencrypted just try to find unencrypted ubuntu-save and then
-	// maybe mount it
-	return m.locateUnencryptedSave, nil
-}
-
-func (m *stateMachine) locateUnencryptedSave() (stateFunc, error) {
-	part := m.degradedState.partition("ubuntu-save")
-	partUUID, findErr := m.disk.FindMatchingPartitionUUID("ubuntu-save")
-	if err := m.setFindState("ubuntu-save", partUUID, findErr, false); err != nil {
-		return nil, nil
-	}
-	if part.FindState != partitionFound {
-		if part.FindState == partitionNotFound {
-			// this is ok, ubuntu-save may not exist for
-			// non-encrypted device
-			part.MountState = partitionAbsentOptional
-		}
-		// all done, nothing left to try and mount, even if errors
-		// occurred
-		return nil, nil
-	}
-
-	// we found the unencrypted device, now mount it
-	return m.mountSave, nil
+	// if we are unencrypted then we shouldn't try to unlock/mount save with the
+	// run key, instead we want to mount it unencrypted, which only works with
+	// the fallback key path which calls
+	// secboot.UnlockVolumeUsingSealedKeyIfEncrypted which works for both
+	// encrypted and unencrypted disks
+	return m.unlockSaveFallbackKey, nil
```

Comment:
do we have a sanity check that if the data was not encrypted then we fail somehow if save is ? now that we use that even for non encrypted save case?

Reply:
no we do not yet have such a check, I will add one

Reply:
I added one and added a unit test for this specific scenario, this situation causes us to "stop the line"